src/main/build/build_config.h

```diff
@@ -68,4 +68,11 @@ typedef enum {
     MCU_TYPE_UNKNOWN = 255,
 } mcuTypeId_e;
 
+typedef struct mcuTypeInfo_s {
+   const mcuTypeId_e id;
```

Comment:
(`const char* name;` is not-const pointer to const string. Applying const to whole struct should do)
```suggestion
   mcuTypeId_e id;
```